tests/system/tests/test_groupdel.py

```diff
@@ -0,0 +1,30 @@
+"""
+Test groupdel
+"""
+
+from __future__ import annotations
+
+import pytest
+
+from framework.roles.shadow import Shadow
+from framework.topology import KnownTopology
+
+
+@pytest.mark.topology(KnownTopology.Shadow)
+def test_groupdel__delete_group(shadow: Shadow):
+    """
+    :title: Basic group deletion
+    :setup:
+        1. Create group
+        2. Delete group
+    :steps:
+        1. Group doesn't exist
+    :expectedresults:
+        1. Group is not found
+    :customerscenario: False
+    """
+    shadow.groupadd("tgroup")
+    shadow.groupdel("tgroup")
+
+    result = shadow.tools.getent.group("tgroup")
```

Comment:
Now, arguably the things being tested for here are not the same.  The original test checks for no changes at all to passwd, shadow etc.  Theoretically if groupadd or groupdel wrongly made a change to another entry in the db, the old test would find it and the new would not.

At first that felt like an improvement (more precision in the verification step), but I worry that a dumb string parsing snafu in some complicated patch could in fact cause something like this.

What do you think?

Reply:
I see your point here, and that would apply to all tests, right? I mean, `useradd` could also be doing something wrong to another entry of the db when adding a new user.

In general terms, do we want to check it? This would add some complexity to all the tests, but I think it is manageable. Database management APIs are stable (and have been stable for a long time), so the key question would be, how likely is this to happen?

Reply:
What I like about your version is it makes clear what we are testing for.

Most of the tests should result in most of the shadow related files not being modified, right?  Perhaps we should actually tell the testsuite which files should change, and the testsuite always checks for changes in any others?

Reply:
I'm not *quite* sure how different that is, so I might be misunderstanding, but it sounds good to me :)